peer/test_utils.go

```diff
@@ -719,3 +720,41 @@ func createTestPeer(t *testing.T) *peerTestCtx {
 		chanStatusMgr: chanStatusMgr,
 	}
 }
+
+func startPeer(t *testing.T, mockConn *mockMessageConn, peer *Brontide) {
+	go func() {
```

Comment:
pls add a comment about why this is being spun off into a goroutine 🙏  and also re why this go routine doesnt need to be tracked (ie, do one of the assertions below mean that this has completed successfully? ie, how do we know this goroutine has been run & has completed?)

Reply:
i think this comment is not addressed fully. 

Should we track this goroutine so we can ensure that it does run or does one of the assertions below mean that function was definitely run?

perhaps to be safe it would anyways be good to add a tracking channel:
something like:


startPeer := make(chan struct{})

then, in the goroutine, after the `.NoError`, add:

close(startPeer)

then wait on this at the end of this function

select{
<-startPeer
<- timeout
}

Reply:
Isn't this redundant? Is there a scenario where init messages are successfully sent and received and this function does not run?

Reply:
Also if we do it this way, we would not be able to test for conditional startup messages such as `peerStorageRetrieval` message in the peer backup PR.

I guess that is what you were asking for in the first place, apologies that I am just addressing it now.

Reply:
im just saying pls add at least a comment like "Because we test that we get message X below, we know that this go-routine does run and does exit since message X is only sent on completion of the Start function"

and if that is _not_ true, then we should add a waitgroup or something. Just good practice.

Reply:
Thanks, I have updated this PR @ellemouton